pkg/querier/querier.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package querier provides tools to interact with the Minder database
+package querier
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/mindersec/minder/internal/db"
+	"github.com/mindersec/minder/pkg/config/server"
+	"github.com/mindersec/minder/pkg/engine/selectors"
+	"github.com/mindersec/minder/pkg/eventer"
+	"github.com/mindersec/minder/pkg/profiles"
+	"github.com/mindersec/minder/pkg/ruletypes"
+)
+
+// Store interface provides functions to execute db queries and transactions
+type Store interface {
+	ProjectHandlers
+	RuleTypeHandlers
+	ProfileHandlers
+	BundleHandlers
+	Close()
+	Commit() error
```

Comment:
Do you need a way to start a transaction, given that you have a `Commit` method?

Reply:
Oh, it looks like you need to throw away the object (and the connection pool) after each transaction?